File: Scenerio18.py
from pyspark import SparkConf, SparkContext
from pyspark.sql import SparkSession
from pyspark.sql import *
from pyspark.sql.types import *
from pyspark.sql.functions import *
from pyspark.sql.window import *

conf = SparkConf().setMaster("local[*]").setAppName("Scenerio18")
sc = SparkContext(conf=conf)
sc.setLogLevel("ERROR")
spark = SparkSession.builder.getOrCreate()

# Create input DataFrame
inputdf = spark.createDataFrame([("The Social Dilemma",)], ["word"])
inputdf.show()

# Define UDF for reversing words
def reverse_sentence(sentence):
    return " ".join([word[::-1] for word in sentence.split(" ")])

# Register UDF
reverse_udf = udf(reverse_sentence, StringType())

# Apply UDF to input DataFrame
outputdf = inputdf.withColumn("reverse word", reverse_udf("word")).drop("word")
outputdf.show()

File: src/pack/Scenerio18.scala
package pack
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.sql._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.expressions._
object Scenerio18 {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("Scenerio18")
    val sc = new SparkContext(conf)
    sc.setLogLevel("ERROR")
    val spark = SparkSession.builder().getOrCreate()
    import spark.implicits._

    val inputdf = Seq("The Social Dilemma").toDF("word")
    inputdf.show()
    val reverseudf = udf((sentence: String) => sentence.split(" ").map(_.reverse).mkString(" "))
    val outputdf = inputdf.withColumn("reverse word", reverseudf($"word")).drop("word")
    outputdf.show()
  }
}